Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Models/PrimerCardValidationModels.swift

```diff
@@ -42,22 +42,41 @@ public class PrimerCardNetwork: NSObject {
         self.displayName = displayName
         self.network = network
     }
+    
+    convenience init(network: CardNetwork) {
+        self.init(
+            displayName: network.validation?.niceType ??
+                network.rawValue.lowercased().capitalized.replacingOccurrences(of: "_", with: " "),
+            network: network
+        )
+    }
 }
 
 @objc
-public class PrimerCardNumberEntryMetadata: NSObject, PrimerPaymentMethodMetadata {
-        
-    public var preferredCardNetwork: PrimerCardNetwork? {
-        return availableCardNetworks.first
+public class PrimerCardNetworksMetadata: NSObject {
+    let items: [CardNetwork]
+    let preferred: CardNetwork?
+    
+    init(items: [CardNetwork], preferred: CardNetwork?) {
+        self.items = items
+        self.preferred = preferred
     }
+}
+
+@objc
+public class PrimerCardNumberEntryMetadata: NSObject, PrimerPaymentMethodMetadata {
     
     public let source: PrimerCardValidationSource
 
-    public let availableCardNetworks: [PrimerCardNetwork]
+    public let selectableCardNetworks: [PrimerCardNetwork]?
```

Comment:
[q]: should this be `PrimerCardNetworksMetadata?`

Reply:
Instead of [this](https://github.com/primer-io/primer-sdk-ios/pull/780/files/0c5f6f6aaf8f42fb1126e493143cf43cefb31b0b#diff-53ee7ceffa5e45b36e07fe4219621f41ef2ed9334aa42c6e2508c644a9c8e153R56)? 

Not sure I follow - or do you mean rename to `PrimerCardNetworkMetadata` (no `s`) ?

Reply:
`data class PrimerCardNumberEntryMetadata(
    val selectableCardNetworks: CardNetworksMetadata?,
    val detectedCardNetworks: CardNetworksMetadata,
    val source: ValidationSource
) : PrimerPaymentMethodMetadata`

something like this